### ISSUE TITLE
Ingress VMSwitch bound to wrong ethernet adapter

### ISSUE THREAD

**Issue by gh2k** [docker/for-win#1399]
On a machine with 2 ethernet adapters, one with e.g. 192.168.10.1/24 and the other with 192.168.11.1/24, joining a swarm with 192.168.11.2 as master:

```
docker swarm join  --token x --advertise-addr 192.168.11.1 --listen-addr 192.168.11.1 192.168.11.2
```
and a service created, such as:
```
docker service create --name foo --replicas 2 --publish 8080:8080 foo/bar"
```

### Expected behavior

Overlay mesh networking should work. Service should be available on 192.168.11.1:8080. Containers should be able to ping each other.

### Actual behavior

None of the above.

### Information

  - Ingress VMSwitch is created bound to the wrong ethernet adapter. For example:

```
# docker network inspect ingress

[
    {
        "Name": "ingress",
        "Id": "y3482d8uq1x3riyy8y5bpmrdu",
        "Created": "2017-12-08T08:05:05.7052246-08:00",
        "Scope": "swarm",
...
```
IP address bound to adapter ethernet1:
```
# Get-NetIPAddress -IPAddress 192.168.11.1

IPAddress         : 192.168.11.1                          
InterfaceIndex    : 4                                                           
InterfaceAlias    : Ethernet1                                                   
AddressFamily     : IPv4                 
...
```
VMSwitch bound to adapter with description "vmxnet3 Ethernet Adapter":
```
Get-VMSwitch -Name y3482d8uq1x3riyy8y5bpmrdu               

Name                      SwitchType NetAdapterInterfaceDescription             
----                      ---------- ------------------------------             
y3482d8uq1x3riyy8y5bpmrdu External   vmxnet3 Ethernet Adapter           
```
Net adapter description for ethernet 1 is *not* the above:
```
# Get-NetAdapter

Name                      InterfaceDescription                    ifIndex Statu 
                                                                          s     
----                      --------------------                    ------- ----- 
vEthernet (Ethernet0 2)   Hyper-V Virtual Ethernet Adapter #2          14 Up    
vEthernet (nat)           Hyper-V Virtual Ethernet Adapter              2 Up    
Ethernet1                 Intel(R) PRO/1000 MT Network Conne...#2       4 Up    
Ethernet0 2               vmxnet3 Ethernet Adapter                      6 Up    
```

I think the VMSwitch for the ingress network should be bound to adapter Ethernet1, but it isn't. I believe this to be the cause of my problems above.

**Comment by gh2k** [comment]
Some further info after some playing.

* Adding a value for `--data-path-addr` to the `docker join` command does not help.
* Using `Ethernet1` instead of an IP address for the `*-addr` parameters on `docker join` does not help.
* Changing `hosts` in `daemon.json` to only bind to the Ethernet1 IP instead of `0.0.0.0` does not help.
* When using the same setup, if I only have *one network adapter* then **it works** just fine, as one might expect from reading the docs

I've tried manually assigning the correct network adapter to the ingress vmswitch post-join, by saying something along the lines of:
```
Set-VMSwitch -Name $vmswitch -InterfaceAlias Ethernet1
New-NetIPAddress -InterfaceAlias "vEthernet ($vmswitch)" -IPAddress $the_ip_i_want -PrefixLength 24
```
That causes more stuff to break and I start getting HNS errors. Restarting Docker appears to delete the switch and recreate it, and I'm back to square one:
```
PS C:\Users\vagrant> Get-VMSwitch

Name                      SwitchType NetAdapterInterfaceDescription
----                      ---------- ------------------------------
nat                       Internal
kmf56gdpgz7e1t8sm5dgkol5e External   Intel(R) PRO/1000 MT Network Connection #2


PS C:\Users\vagrant> Stop-Service docker
PS C:\Users\vagrant> Start-Service docker
PS C:\Users\vagrant> Get-VMSwitch

Name                      SwitchType NetAdapterInterfaceDescription
----                      ---------- ------------------------------
nat                       Internal
kmf56gdpgz7e1t8sm5dgkol5e External   vmxnet3 Ethernet Adapter
```

Basically, I can't get overlay networking working *at all* if I want to use Docker on my second network adapter.

**Comment by rgl** [comment]
any news about this?

**Comment by rgl** [comment]
ping?

**Comment by drnybble** [comment]
I put a fix into the issue linked above
